src/core/ext/client_channel/client_channel.c

```diff
@@ -336,6 +336,7 @@ static void on_resolver_result_changed(grpc_exec_ctx *exec_ctx, void *arg,
         exec_ctx, chand, state, GRPC_ERROR_REF(state_error), "new_lb+resolver");
     if (lb_policy != NULL) {
       watch_lb_policy(exec_ctx, chand, lb_policy, state);
+      GRPC_LB_POLICY_WEAK_REF(lb_policy, "on_resolver_results_changed");
```

Comment:
I've 's/lb_policy/chand->lb_policy`'d to emphasize that we are taking a ref on something the `on_resolver_results_changed` expects.